src/main/java/seedu/data/ingredient/Ingredient.java

```diff
@@ -53,7 +53,7 @@ public Ingredient(String ingredientName, int quantity) {
      * @param qty Amount of items to be added.
      */
     public void addQuantity(int qty) {
-        quantity += qty;
+        quantity += (quantity ==  Integer.MAX_VALUE) ? 0 : qty;
```

Comment:
Shouldn't we catch for all numbers greater than or equal to Integer.MAX_VALUE or less than or equal to Integer.MIN_VALUE?

Reply:
Actually shouldn't the check be if `quantity + qty >= Integer.MAX_VALUE ? 0 : qty`

Reply:
you can't be greater than MAX if it overflows.

Reply:
I could change it to `(quantity + qty < 0) ? 0:qty` but this makes catching the exception difficult. because we don't throw exceptions in command classes.